### PR TITLE
backport: v0.20 pr9

### PR DESCRIPTION
**Pull request by Munkybooty**
## Issue being fixed or feature implemented

Bitcoin Backports

## What was done?

Misc v20 Backports, check commit log to see changes.

## How Has This Been Tested?

Compilation and tested unit tests locally

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -576,7 +576,7 @@ static UniValue getrawmempool(const JSONRPCRequest& request)
                 {
                     {RPCResult::Type::STR_HEX, "", "The transaction id"},
                 }},
-            RPCResult{"for verbose = true",
+                    RPCResult{"for verbose = true",
```
[the remark this answers is not in the document]
Comment (Munkybooty) [reply src/rpc/blockchain.cpp:579]: if I remember correctly, all those changes may have already been added in a different pr

**Comment by knst** [comment]
@Munkybooty please, check this commit

File: src/test/util/transaction_utils.cpp [hunk 2]
```diff
@@ -36,3 +38,33 @@ CMutableTransaction BuildSpendingTransaction(const CScript& scriptSig, const CTr
 
     return txSpend;
 }
+
+std::vector<CMutableTransaction> SetupDummyInputs(FillableSigningProvider& keystoreRet, CCoinsViewCache& coinsRet, const std::array<CAmount,4>& nValues)
```
Comment (knst) [inline src/test/util/transaction_utils.cpp:42]: I see added new function `SetupDummyInputs` but don't see removed old copies in:
 - src/bench/ccoins_caching.cpp
 - src/test/transaction_tests.cpp

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -2255,7 +2255,7 @@ static UniValue listlockunspent(const JSONRPCRequest& request)
                     {RPCResult::Type::STR_HEX, "txid", "The transaction id locked"},
                     {RPCResult::Type::NUM, "vout", "The vout value"},
                 }},
-            }
+                    }
```
Comment (knst) [inline src/wallet/rpcwallet.cpp:2258]: Looks like formatting is broken now, used to be better before... Is it expecting? 
And not only there, but in other places.

**Review by knst — CHANGES_REQUESTED** [review]
bitcoin 17959 - LGTM
bitcoin 18173 - check comments
bitcoin 18253 - LGTM
bitcoin 18263 - LGTM
bitcoin 18306 - LGTM
bitcoin 15443 - LGTM
bitcoin 17809 - seems as formatting become worse and broken now in many places. Used to be better. Btw, I will review later if any missing useful changes from 17809.

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**